src/signals/incident-management/containers/IncidentDetail/components/Attachments/EditAttachment..tsx

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (C) 2023 Gemeente Amsterdam
+import type { ReactNode } from 'react'
+import { useCallback } from 'react'
+
+import { Button, Checkbox, Label } from '@amsterdam/asc-ui'
+import { yupResolver } from '@hookform/resolvers/yup/dist/yup'
+import { useForm, Controller } from 'react-hook-form'
+import * as yup from 'yup'
+
+import { ButtonsWrapper, EditAttachmentWrapper, StyledForm } from './styled'
+import TextArea from '../../../../../../components/TextArea'
+import type { Attachment } from '../../types'
+import { StyledH2 } from '../StatusForm/styled'
+
+type Props = {
+  attachment: Attachment
+  children: ReactNode
+  editActive: boolean
+  setEditActive: (location: string | null) => void
+  patch: (href: string, attachment: FormValues) => void
+}
+
+type FormValues = {
+  public: boolean
+  caption?: string
+}
+
+export default function EditAttachment({
+  attachment,
+  children,
+  patch,
+  editActive,
+  setEditActive,
+}: Props) {
+  const schema = yup.object().shape({
+    public: yup.boolean(),
+    caption: yup
+      .string()
+      .nullable()
+      .max(120, 'Je hebt meer dan de maximale 120 tekens ingevoerd.'),
+  })
+
+  const { handleSubmit, formState, control, getValues } = useForm<FormValues>({
+    defaultValues: {
+      public: attachment.public,
+      caption: attachment.caption,
+    },
+    resolver: yupResolver(schema),
+  })
+
+  const { errors, dirtyFields } = formState
+
+  const onSubmit = useCallback(
+    (data) => {
+      const dirtyKeys = Object.keys(dirtyFields)
+      let dirtyData = dirtyKeys.reduce((acc, key) => {
+        if (data[key] !== undefined) {
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          acc[key] = data[key]
+        }
+        return acc
+      }, {} as FormValues)
+      if (!data.public && dirtyFields.public && dirtyData.caption) {
+        dirtyData = { ...dirtyData, caption: null } as FormValues & {
+          caption: null
+        }
+      }
```

Comment:
What's happening here?

Reply:
the null value can't be added through the form. This is a bit like patching the data before sending the PATH request.